parts/windows/kuberneteswindowssetup.ps1

```diff
@@ -444,6 +451,23 @@ try
         Install-GmsaPlugin -GmsaPackageUrl $global:WindowsGmsaPackageUrl
     }
 
+    Write-Log "Stage1 completed successfully"
+    Logs-To-Event -TaskName "AKS.WindowsCSE.Stage1" -TaskMessage "Stage1 completed successfully"
+}
+
+# ====== STAGE 2: CLUSTER INTEGRATION ======
+# All operations that should only run when connecting to the actual cluster
+function Stage2 {
+    if ($PreProvisionOnly) {
```

Comment:
I have a preference to have this logic in the place that invokes the State1 or Stage2 function as then the "stage control flow" is all in one spot.  I was confused when reading the bit that invokes stage1 and stage2 that we seemed to always invoke them and had to read the individual methods to see that control flow.

Reply:
updated